File: backend/src/models/LeagueSettings.ts
import { Schema, model } from "mongoose";

export interface LeagueSettingsAttrs {
  resultEntryDelayMinutes: number;
  matchWinPoints: number;
  tournamentWinPoints: number;
  inactivityDays: number;
}

const leagueSettingsSchema = new Schema<LeagueSettingsAttrs>(
  {
    resultEntryDelayMinutes: { type: Number, default: 60, min: 0 },
    matchWinPoints: { type: Number, default: 10, min: 0 },
    tournamentWinPoints: { type: Number, default: 50, min: 0 },
    inactivityDays: { type: Number, default: 0, min: 0, max: 365 }
  },
  {
    timestamps: true,
    toJSON: {
      transform(_doc, ret) {
        delete (ret as { __v?: number }).__v;
        return ret;
      }
    }
  }
);

export const LeagueSettings = model<LeagueSettingsAttrs>("LeagueSettings", leagueSettingsSchema);

File: lib/models/league_settings.dart
class LeagueSettings {
  const LeagueSettings({
    required this.resultEntryDelayMinutes,
    required this.matchWinPoints,
    required this.tournamentWinPoints,
    this.inactivityDays = 0,
  });

  final int resultEntryDelayMinutes;
  final int matchWinPoints;
  final int tournamentWinPoints;
  final int inactivityDays;

  factory LeagueSettings.fromJson(Map<String, dynamic> json) {
    return LeagueSettings(
      resultEntryDelayMinutes: json['resultEntryDelayMinutes'] ?? 60,
      matchWinPoints: json['matchWinPoints'] ?? 10,
      tournamentWinPoints: json['tournamentWinPoints'] ?? 50,
      inactivityDays: json['inactivityDays'] ?? 0,
    );
  }
}
